asn1sccserviceprovider.cpp

```diff
@@ -0,0 +1,152 @@
+/****************************************************************************
+**
+** Copyright (C) 2017 N7 Mobile sp. z o. o.
+** Contact: http://n7mobile.com/Space
+**
+** This file is part of ASN.1/ACN Plugin for QtCreator.
+**
+** Plugin was developed under a programme and funded by
+** European Space Agency.
+**
+** This Plugin is free software: you can redistribute it and/or modify
+** it under the terms of the GNU General Public License as published by
+** the Free Software Foundation, either version 3 of the License, or
+** (at your option) any later version.
+**
+** This Plugin is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+** GNU General Public License for more details.
+**
+** You should have received a copy of the GNU General Public License
+** along with this program.  If not, see <http://www.gnu.org/licenses/>.
+**
+****************************************************************************/
+
+#include "asn1sccserviceprovider.h"
+
+#include <QString>
+#include <QSettings>
+
+#include <QJsonArray>
+#include <QJsonObject>
+
+#include <coreplugin/icore.h>
+
+#include "asn1acnconstants.h"
+
+using namespace Asn1Acn::Internal;
+
+const int STAY_ALIVE_PERIOD_MS = 500;
+
+Q_GLOBAL_STATIC(QNetworkAccessManager, networkManagerInstance)
+
+Asn1SccServiceProvider::Asn1SccServiceProvider() :
+    m_asn1sccService(new QProcess(this)),
+    m_terminating(false)
+{
+    loadServiceParams();
+
+    m_asn1sccService->setProgram(m_serviceBinaryPath);
+    m_asn1sccService->setArguments(additionalArguments());
+
+    // QProcess::finished is overloaded, hence the cast
+    connect(m_asn1sccService, static_cast<void(QProcess::*)(int, QProcess::ExitStatus)>(&QProcess::finished),
+            this, &Asn1SccServiceProvider::onProcessFinished);
+
+    m_stayAliveTimer.setInterval(STAY_ALIVE_PERIOD_MS);
+
+    connect(&m_stayAliveTimer, &QTimer::timeout,
+            this, &Asn1SccServiceProvider::stayAliveTimeout, Qt::UniqueConnection);
+}
+
+Asn1SccServiceProvider::~Asn1SccServiceProvider()
+{
+    if (m_asn1sccService->state() != QProcess::NotRunning)
+        finalize();
+}
+
+QNetworkReply *Asn1SccServiceProvider::requestAst(const QString &documentData, const QFileInfo &fileInfo) const
```

Comment:
it still returns network related classes - maybe some signals should be added to Asn1SccServiceProvider, so `requestAst` returns `void` but there's some signal like `void astReceived()`